pkg/sync/syncer.go

```diff
@@ -1974,7 +1974,7 @@ func (s *syncer) processGrantsWithExternalPrincipals(ctx context.Context, princi
 						l.Error("error getting user trait", zap.Any("userPrincipal", userPrincipal))
 						continue
 					}
-					if matchExternalResource.Key == "email" {
+					if strings.EqualFold(matchExternalResource.Key, "email") {
 						if userTraitContainsEmail(userTrait.Emails, matchExternalResource.Value) {
```

Comment:
I think you'll need to mess with some strings in userTraitContainsEmail() to get emails to match if they differ in case.

Reply:
I used `EqualFold` in `userTraitContainsEmail`